cpp/ql/src/semmle/code/cpp/Type.qll

```diff
@@ -1308,7 +1308,7 @@ class SpecifiedType extends DerivedType {
    * only depends on the specifiers, not on the source program). This is intended
    * for debugging queries only and is an expensive operation.
    */
-  string getSpecifierString() { internalSpecString(this, result, 1) }
+  string getSpecifierString() { result = concat(this.getASpecifier().getName(), " ") + " " }
```

Comment:
How do you feel about removing the trailing space, and fixing up the three? places this predicate is used?

Reply:
We could do that, but in theory it is an API change.
It is rectified by the fact that the predicate is only for debugging, and that the whitespace behaviour is not documented, but still, at least in my opinion it's not a no-brainer.

Reply:
I'm happy to accept this API change in a predicate that has "This is intended for debugging queries only" in its QLDoc.

This is a good reminder that we should be conscious about which APIs we make public. The QLDoc for `getSpecifierString` should really start with "INTERNAL: do not use.", and ideally that predicate should not reside in `Type.qll` at all.